src/Controller/AbstractSecurityController.php

```diff
@@ -43,16 +40,115 @@ public function loginAction()
             'error'         => $helper->getLastAuthenticationError(),
             'layout_template' => $this->context . '/empty_layout.html.twig',
             'security_login_check_url' => $this->generateUrl($this->context . '_security_login_check'),
-//            'security_forgot_password_url' => $this->generateUrl($this->context . '_security_forgot_password'),
+            'security_forgot_password_url' => $this->generateUrl($this->context . '_security_forgot_password'),
         ]);
     }
 
     /**
-     * @return AuthenticationUtils
+     * @param Request $request
+     *
+     * @return Response
      */
-    private function getAuthenticationUtils()
+    public function forgotPasswordAction(Request $request)
     {
-        return $this->get('security.authentication_utils');
+        $form =  $this->createForm(ForgotPasswordType::class);
+        $form->handleRequest($request);
+
+        if (!$form->isSubmitted() || !$form->isValid()) {
+            return $this->render(
+                $this->context . '/security/forgot_password.html.twig',
+                [
+                    'form' => $form->createView(),
+                    'security_login_form_url' => $this->generateUrl($this->context . '_security_login_form'),
+                    'security_forgot_password_url' => $this->generateUrl($this->context . '_security_forgot_password'),
+                ]
+            );
+        }
+
+        $user = $this->get($this->context . '_user_provider')->loadUserByUsername($form->get('email')->getData());
+
+        $this->addFlash('success', 'flash.forgot_password.success');
+
+        if ($user) {
+            $token = $this->getTokenManager()->create(Token::RESET_PASSWORD, $user);
+
+            $this->getMessenger()->send(
+                'security.forgot_password',
+                $user,
+                [
+                    '{context}' => $this->context,
+                    'token' => $token->getValue(),
+                    'domain' => $this->container->getParameter('domain'),
+                    'security_reset_password_route' => $this->context . '_security_reset_password'
+                ]
+            );
+        }
```

Comment:
You will still get the flash message even if you do not find the user with `loadUserByUsername`

Reply:
Yes this is done on purpose so nobody can check if user exist or not on or application

Reply:
clever ;)